Update: GitHub Actions workflow to remove the bot comment when Trac Ticket is added to the PR.

Debarghya-Banerjee opened the pull request:
**Trac Ticket - [Core-61567](https://core.trac.wordpress.org/ticket/61567)**

## Problem Statement:
Bot comments reminding contributors to include Trac ticket links in pull requests accumulate over time, cluttering discussion threads even after the links are added. This clutter hinders efficient review processes and detracts from meaningful collaboration in WordPress core development.

## Solution:

- Implement automated checks to:
    - Verify each pull request includes a link to a corresponding Trac ticket.
    - Create a notice if a Trac ticket link is missing.
    - Automatically remove the notice upon adding a valid Trac ticket link to streamline pull request review processes and reduce unnecessary bot comments.

## Impact:

- Automating the deletion of bot comments once the Trac ticket link is updated in GitHub pull requests for WordPress core development provides several key benefits:

    - `Enhanced Review Efficiency`: Streamlines PR discussions by reducing clutter, allowing faster decision-making and clearer communication.
    - `Improved Collaboration`: Encourages focused participation and feedback, promoting a more engaged development process.
    - `Simplified Workflow`: Enhances user experience by minimizing distractions and optimizing contributor workflows.
    - `Overall Efficiency`: Optimizes reviewer resources, maximizing productivity and the impact of community contributions.

Commit:
Update github actions workflow to remove the bot comment when Trac Ticket gets updated

desrosj commented on .github/workflows/pull-request-comments.yml:
```diff
@@ -164,16 +164,16 @@ jobs:
+  # Leaves a comment on a pull request when no Trac ticket is included in the pull request description and delete the comment if the Trac ticket has been updated.
   trac-ticket-check:
-    name: Comment on a pull request when no Trac ticket is included
+    name: Comment on a pull request when no Trac ticket is included and delete the comment when Trac Ticket has been updated
```
Let's adjust this a bit. It's possible that more tasks are added to this in the future. So let's be more general for the `name` and move the details into the description.

Debarghya-Banerjee replied:
Fixed

desrosj commented on .github/workflows/pull-request-comments.yml:
```diff
@@ -164,16 +164,16 @@ jobs:
+  # Leaves a comment on a pull request when no Trac ticket is included in the pull request description and delete the comment if the Trac ticket has been updated.
   trac-ticket-check:
-    name: Comment on a pull request when no Trac ticket is included
+    name: Comment on a pull request when no Trac ticket is included and delete the comment when Trac Ticket has been updated
```
```suggestion
    name: Manage Trac ticket reminders for pull requests
```

Debarghya-Banerjee replied:
Fixed

desrosj commented on .github/workflows/pull-request-comments.yml:
```diff
@@ -164,16 +164,16 @@ jobs:
             github.rest.issues.createComment( commentInfo );
 
-  # Leaves a comment on a pull request when no Trac ticket is included in the pull request description.
+  # Leaves a comment on a pull request when no Trac ticket is included in the pull request description and delete the comment if the Trac ticket has been updated.
```
```suggestion
  # Manages comments reminding contributors to include a Trac ticket link when opening a pull request.
  # 
  # This job will:
  # - Leave a comment on a pull request when no Trac ticket is included in the pull request description.
  # - Delete the comment once a Trac ticket has been added.
```

Debarghya-Banerjee replied:
Fixed

Debarghya-Banerjee replied:
Fixed

desrosj reviewed (changes requested):
Thanks @Debarghya-Banerjee! I left a few small suggestions!

I've also been thinking through whether adding any event triggers would be appropriate. `pull_request_target` is required to ensure the workflow has the necessary capabilities to [manage comments and has the following `types`  by default](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target): `opened`, `synchronize`, and `reopened`.

We could probably add `edited` to this list. But all three would need to be explicitly listed.

Debarghya-Banerjee committed:
Merge branch 'trunk' into update/remove-bot-comment-when-tra-ticket-is-being-updated-by-the-user

Debarghya-Banerjee committed:
Address the feedback

Debarghya-Banerjee committed:
Add 'synchronize', 'reopened', and 'edited'

Debarghya-Banerjee commented:
Hi @desrosj, I have addressed the feedback and made the changes. I have also added the `synchronize`, `reopened`, & `edited` in the `pull_request_track`

I am re-assigning you for review. Thanks.

desrosj reviewed (approved):
Thanks @Debarghya-Banerjee! Let's give it a try!

desrosj commented:
Merged in https://core.trac.wordpress.org/changeset/59169.
